spec/system/user_post_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'rails_helper'
+
+describe 'User post index page', type: :feature do
```

Comment:
- [ ] **[OPTIONAL]** You have added the spec tests for the post index and show pages in this one file. I would recommend you split them into two files for better code organization.